src/main/credentialServer/httpCredentialHelperServer.ts

```diff
@@ -143,43 +183,76 @@ export class HttpCredentialHelperServer {
+      pathVar += `${ path.delimiter }/usr/local/bin`;
     }
 
-    try {
-      const platform = os.platform();
-      let pathVar = process.env.PATH ?? '';
-
-      // The PATH needs to contain our resources directory (on macOS that would
-      // not be in the application's PATH), as well as /usr/local/bin.
-      // NOTE: This needs to match DockerDirManager.
-      pathVar += path.delimiter + path.join(paths.resources, platform, 'bin');
-      if (platform === 'darwin') {
-        pathVar += `${ path.delimiter }/usr/local/bin`;
-      }
+    const body = stream.Readable.from(data);
+    const { stdout } = await childProcess.spawnFile(helperName, [commandName], {
+      env:   { ...process.env, PATH: pathVar },
+      stdio: [body, 'pipe', console],
+    });
 
-      const body = stream.Readable.from(data);
-      const { stdout } = await childProcess.spawnFile(helperName, [commandName], {
-        env:   { ...process.env, PATH: pathVar },
-        stdio: [body, 'pipe', console],
-      });
+    if (!checkerFn(stdout)) {
+      throw new Error(`Invalid output for ${ commandName } command.`);
+    }
 
-      if (checkerFn(stdout)) {
-        response.writeHead(200, { 'Content-Type': 'text/plain' });
-        response.write(stdout);
+    return stdout;
+  }
 
-        return;
+  /**
+   * For the LIST command, there are multiple possible sources of information
+   * that need to be merged into a simple
+   *    { ServerURL: Username } hash.
+   * The first source is the credsStore.
+   * Then if any helper credsStores are identified in the `credHelpers` section,
+   * get the full { ServerURL: Username } from each of them,
+   * and keep only those ServerURLs that point to that credsStore.
+   *
+   * Modeled after https://github.com/docker/cli/blob/d0bd373986b6678bfe1a0eb6989ce13907247a85/cli/config/configfile/file.go#L285
+   */
+  protected async doListCommand(
+    thisHelperInfo: credHelperInfo,
+    request: http.IncomingMessage,
+    response: http.ServerResponse): Promise<void> {
+    try {
+      const serverAndUsernameInfo: Record<string, string> = JSON.parse(await this.runCommand(`docker-credential-${ thisHelperInfo.credsStore }`, 'list', '', request));
+      const names = _.uniq(Object.values(thisHelperInfo.credHelpers ?? {}));
+
+      for (const name of names) {
+        try {
+          const otherInfo = JSON.parse(await this.runCommand(`docker-credential-${ name }`, 'list', '', request));
+
+          for (const [serverURL, username] of Object.entries(otherInfo)) {
+            if (thisHelperInfo.credHelpers[serverURL] === name) {
+              serverAndUsernameInfo[serverURL] = username as string;
+            }
+          }
+        } catch (err) {
+          console.debug(`Failed to get credential list for helper ${ name }`);
+        }
       }
-      stderr = stdout;
+      response.writeHead(200, { 'Content-Type': 'text/plain' });
+      response.write(JSON.stringify(serverAndUsernameInfo));
+
+      return;
```

Comment:
[Unused `return`](https://eslint.org/docs/latest/rules/no-useless-return) (that we're not linting for).

Reply:
Weird that the linter isn't using the ASL to see unnecessary `return`s like that one.